storage/innobase/handler/ha_innodb.cc

```diff
@@ -14476,8 +14476,8 @@ ha_innobase::records_in_range(
 		n_rows = rtr_estimate_n_rows_in_range(
 			index, range_start, mode1);
 	} else {
-		btr_pos_t tuple1(range_start, mode1, pages->first_page);
-		btr_pos_t tuple2(range_end,   mode2, pages->last_page);
+		btr_pos_t tuple1(range_start, mode1, 0);
+		btr_pos_t tuple2(range_end,   mode2, 0);
```

Comment:
This would seem to be the actual fix. `~0ULL` might be a safer value, but I think that 0 should be OK as well, because the smallest possible index page number is 3.

Reply:
I think I could add `MEM_UNDEFINED(&tuple1.page_id, sizeof tuple1.page_id)` (plus tuple2) to protect against future `btr_estimate_n_rows_in_range` that access it.